### DISCUSSION TITLE
important hyperparameters for graph attention network

### DISCUSSION THREAD

**Discussion by harirRzi** [pyg-team/pytorch_geometric#9203] [Q&A]
Hello. I have multiple graphs and I want to train on some part of them and then test on the rest of graphs inorder to predict node labels.I have implemented a Graph Attention network. Here is what I have implemented: 

```
class GraphAtt(torch.nn.Module):

  def __init__(self, in_channels, hidden_channels, out_channels):
    
    super().__init__()

    heads = 1
    self.gat1 = GATConv(in_channels, hidden_channels, heads=heads, dropout=0.5)
    self.norm1 = torch.nn.LayerNorm(hidden_channels * heads)
    self.gat2 = GATConv(hidden_channels * heads, out_channels, heads=1, dropout=0.5)
    self.norm2 = torch.nn.LayerNorm(out_channels)

  def forward(self, x, edge_index, edge_weight=None):
        x = self.gat1(x, edge_index, edge_weight)
        x = self.norm1(x)
        x = F.relu(x)
        x = F.dropout(x, p=0.5, training=self.training)
        
        x = self.gat2(x, edge_index, edge_weight)
        x = self.norm2(x)
        x = F.elu(x)
        x = F.dropout(x, p=0.5, training=self.training)
        
        out = F.log_softmax(x, dim=1)
        label = out.argmax(dim=-1)
        return label, out 
``` 
        
After I train the model and test on some of graphs, I see the performance of it is not good. I want to know if there is any specific setting or parameters needed for graph attention network that can make its performance to become better?

**Comment by rusty1s** [comment]
You shouldn't use `norm` and `activation before `softmax`. Try to add a `Linear` layer before `log_softmax` call.
